src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp

```diff
@@ -107,6 +107,16 @@ ov::pass::ConvertGatherToGatherCompressed::ConvertGatherToGatherCompressed() {
         std::shared_ptr<ov::Node> gather_input_zp = optional_zero_point;
         std::vector<std::shared_ptr<ov::Node>> result_nodes = {};
 
+        ov::element::Type output_type = gather_node->get_output_element_type(0);
```

Comment:
The `Convert` of original model must be kept here for graph compatibility under different runtime precision. The `Convert` here keeps the original precision of `Scale`, If we remove it, the model will lose the flexibility to run different precision. @xipingyan @e-ddykim

Reply:
I'm not sure about the problem to be fixed by this PR. If the original output type and scale type are different, does this mean that the scale type should take priority?

Reply:
The problem happens when the FP32 model is quantized to INT8, the scale here is used to do the quantization. The `Convert` is inserted by `NNCF`. I think this `Convert` should be kept and later handled by LPT transformations. If we remove the `Convert` here, The quantization pattern is broken.
![image](https://github.com/openvinotoolkit/openvino/assets/47727876/28f67af6-906c-47f2-b2c7-284733ef1fb9)

Reply:
Thank you for the capture. But I don't understand what the problem is. Is the output type of `Gather_1070` fp16? Even if so, I don't understand the reason to change the output type of `Gather_1070` to fp32. Additionally, why is the output type of `Gather_1070` fp16 in the fp32 model? I'm sorry for many questions, but I'm quite confused.

Reply:
Hi, let me describe the problem from beginning.

1. The `convert_gather_to_gather_compressed` converts `Gather_1070` to `GatherCompressed` and set its output to `FP32`
2. When user sets `infer_precision` to `fp16`, the output of `GatherCompressed` cannot be changed to `fp16` because `m_output_type` can only configured at class construction phase.
3. Then I change the `GatherCompressed` to follow the actual precision which is the same precision of the scale input
```
output_type = get_input_element_type(3)
```
4. Step 3 fixes the problem in bert-like model. However it triggers another problem in LLM-int8 model. The output of `GatherCompressed`(Gahter_1070) is connect to `Power_1075` which has other input `Convert_1074`. If we remove the `Convert_1067`, later, the output(FP16) of `GahterCompressed` is not compatible with another input of `Convert_1074`(FP32). Therefore, here I propose to maintain the `Convert_1067`
![image](https://github.com/openvinotoolkit/openvino/assets/47727876/5e0d7c45-1d07-49b2-a82b-cce1343140cf)